test/AutoDiff/validation-test/forward_mode.swift

```diff
@@ -1319,4 +1319,23 @@ ForwardModeTests.test("ForceUnwrapping") {
   expectEqual(5, forceUnwrap(Float(2)))
 }
 
+//===----------------------------------------------------------------------===//
+// Array methods from ArrayDifferentiation.swift
+//===----------------------------------------------------------------------===//
+
+ForwardModeTests.test("Array.differentiableMap") {
+  let a: [Float] = [1, 2, 3]
+  let tan = Array<Float>.TangentVector.init(a)
```

Comment:
Minor: you can avoid spelling `.init` when directly initializing a type (i.e. directly applying a type's initializer):
```suggestion
  let tan = Array<Float>.TangentVector(a)
```

Reply:
There are other cases where explicit `.init` initializer references are useful:
- To directly reference an initializer as a function-value typed value:
```swift
let fn = Array<Float>.init(repeating:count:)
print(type(of: fn)) // (Float, Int) -> Array<Float>
```
- Using `.init` shorthand (an [implicit member expression](https://docs.swift.org/swift-book/ReferenceManual/Expressions.html#ID394)) to refer to a type's initializer, when type inference can determine the implied base type.

Here's an example from an autodiff test:
https://github.com/apple/swift/blob/cb8ea0c8e81f7c7f3b250b8897ad8a0c6f66d06d/test/AutoDiff/validation-test/simple_math.swift#L264-L272

In the example:
- `Foo.init` is a value with type `(Float) -> Foo`.
- `.init(stored: 10)` is shorthand for `Foo.TangentVector.init(stored: 10)`.